File: Osp/src/main/java/de/gso/osp/persistence/model/KalenderEintrag.java
package de.gso.osp.persistence.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "schueler")
public class KalenderEintrag {
	
	@Id
	@Column(unique = true, nullable = false)
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	
	private Date datum;
	
	private String kalenderKey;
	
	private String notiz;
	
	private String unterricht;
	
	private Klasse klasse;
	
	private Benutzer benutzer;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Date getDatum() {
		return datum;
	}

	public void setDatum(Date datum) {
		this.datum = datum;
	}

	public String getNotiz() {
		return notiz;
	}

	public void setNotiz(String notiz) {
		this.notiz = notiz;
	}

	public String getUnterricht() {
		return unterricht;
	}

	public void setUnterricht(String unterricht) {
		this.unterricht = unterricht;
	}

    @ManyToOne
    @JoinColumn(name = "benutzer_id")
	public Benutzer getBenutzer() {
		return benutzer;
	}

	public void setBenutzer(Benutzer benutzer) {
		this.benutzer = benutzer;
	}

    @ManyToOne
    @JoinColumn(name = "klasse_id")
	public Klasse getKlasse() {
		return klasse;
	}

	public void setKlasse(Klasse klasse) {
		this.klasse = klasse;
	}

	public String getKalenderKey() {
		return kalenderKey;
	}

	public void setKalenderKey(String kalenderKey) {
		this.kalenderKey = kalenderKey;
	}

}

File: Osp-PHP/src/Entity/KalenderEintrag.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="App\Repository\KalenderEintragRepository")
 */
class KalenderEintrag
{
    /**
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="datetime")
     */
    private $datum;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $kalenderKey;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $notiz;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Benutzer", inversedBy="kalenderEintrags")
     * @ORM\JoinColumn(nullable=false)
     */
    private $benutzer;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Klasse", inversedBy="kalenderEintrags")
     * @ORM\JoinColumn(nullable=false)
     */
    private $klasse;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Fach", inversedBy="kalenderEintrags")
     * @ORM\JoinColumn(nullable=false)
     */
    private $fach;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDatum(): ?\DateTimeInterface
    {
        return $this->datum;
    }

    public function setDatum(\DateTimeInterface $datum): self
    {
        $this->datum = $datum;

        return $this;
    }

    public function getKalenderKey(): ?string
    {
        return $this->kalenderKey;
    }

    public function setKalenderKey(string $kalenderKey): self
    {
        $this->kalenderKey = $kalenderKey;

        return $this;
    }

    public function getNotiz(): ?string
    {
        return $this->notiz;
    }

    public function setNotiz(string $notiz): self
    {
        $this->notiz = $notiz;

        return $this;
    }

    public function getBenutzer(): ?Benutzer
    {
        return $this->benutzer;
    }

    public function setBenutzer(?Benutzer $benutzer): self
    {
        $this->benutzer = $benutzer;

        return $this;
    }

    public function getKlasse(): ?Klasse
    {
        return $this->klasse;
    }

    public function setKlasse(?Klasse $klasse): self
    {
        $this->klasse = $klasse;

        return $this;
    }

    public function getFach(): ?Fach
    {
        return $this->fach;
    }

    public function setFach(?Fach $fach): self
    {
        $this->fach = $fach;

        return $this;
    }
}
